secret/secret.go

```diff
@@ -61,156 +60,96 @@ func GetSecretInterface() secretInterface {
 	return clientSet.CoreV1().Secrets(getEnv("KUBERNETES_NAMESPACE"))
 }
 
-func UpdateSecrets(s secretInterface, secrets *v1.Secret, create bool) {
-	if create {
-		util.MarkAsClean(secrets)
-		_, err := s.Create(secrets)
-		if err != nil {
-			logFatal(err)
-		}
-		log.Printf("Created `%s`\n", secrets.Name)
-	} else if util.IsDirty(secrets) {
-		util.MarkAsClean(secrets)
-		_, err := s.Update(secrets)
-		if err != nil {
-			logFatal(err)
-		}
-		log.Printf("Updated `%s`\n", secrets.Name)
+func UpdateSecrets(s secretInterface, secrets *v1.Secret) {
+	_, err := s.Create(secrets)
+	if err != nil {
+		logFatal(err)
 	}
+	log.Printf("Created `%s`\n", secrets.Name)
 }
 
-func FindSecret(s secretInterface) (*v1.Secret, error) {
-	slist, err := s.List (metav1.ListOptions{})
-	if err != nil {
-		return nil, err
+func FindPreviousSecret(s secretInterface, rv int) (*v1.Secret, error) {
+	// Args: rv is the current release revision.
+	// Logic
+	// (1) if rv-2 exists, delete it (prevent leaking of too many old entities)
+	// (2) if rv-1 exists take it
+	// (3) if secret without rv exists, take it
+	// (4) error
+
+	// / / // /// ///// //////// ///////////// /////////////////////
+
+	// (1) Delete really old entity (R-2), if it exists. We do not
+	// care about errors.
+	_ = s.Delete(SECRET_NAME + fmt.Sprintf("-%d", rv-2), &metav1.DeleteOptions{})
```

Comment:
I generally prefer to use `fmt.Sprintf` for the complete formatting and not combine it with concatenation, so would write:

```
_ = s.Delete(fmt.Sprintf("%s-%d", SECRET_NAME, rv-2), ...)
```

Reply:
Addressed with commit 8f19e604806707cf83c9957249e323994194767a